src/app/components/button/button.test.js

```diff
@@ -0,0 +1,27 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import Button from './Button';
+
+test('renders correct text', () => {
+  render(<Button buttonText='Click me' />);
+  const buttonElement = screen.getByText(/click me/i);
+  expect(buttonElement).toBeInTheDocument();
+});
+
+test('calls onClick prop when clicked', () => {
+  const handleClick = jest.fn();
+  render(<Button buttonText='Click me' onClick={handleClick} />);
+  fireEvent.click(screen.getByText(/click me/i));
+  expect(handleClick).toHaveBeenCalledTimes(1);
+});
+
+test('has a class when variant is primary', () => {
+  render(<Button buttonText='Primary Button' variant='primary' />);
+  const buttonElement = screen.getByText(/primary button/i);
+  expect(buttonElement).toHaveAttribute('class');
```

Comment:
I'm not sure this test is actually doing what you think it may be doing. Whether the primary variable is applied or not, this button element will also have a class applied to it, correct? This test may be giving you a false-positive.

Reply:
It would need a new test for each variant - generally, though, that's not a bad thing. I think it provides value to know if a specific variant suddenly fails or breaks. Buttons typically have 2-3 variants. Adding a couple of extra tests to our testing suite is manageable imho!

Reply:
is this appropriate? it's telling me it can't find the class

`test('has the correct class when variant is primary', () => {
  render(<Button buttonText='Primary Button' variant='primary' />);
  const buttonElement = screen.getByText(/primary button/i);
  expect(buttonElement).toHaveClass(styles['button--primary']);
});`